src/components/dataset/data/sidebar/filter/CategoryFilterGroup.jsx

```diff
@@ -45,17 +45,24 @@ export class CategoryFilterGroup extends React.PureComponent {
   };
 
   render() {
-    const { column, values, table, originalValues, filterMap } = this.props;
-    const checkboxes = _.keys(originalValues).map((name) => {
-      const count = parseInt(_.get(values, name, '0'), 10);
+    const { column, values, table, filterMap, originalValues } = this.props;
+    const checkboxes = originalValues?.map((val) => {
+      const totalCount = val.count;
+      let filteredCount = 0;
+      values.map((filteredValue) => {
+        if (filteredValue.value === val.value) {
+          filteredCount = filteredValue.count;
+        }
+      });
```

Comment:
(Explanation because this wasn't entirely clear to me before getting into this code)
For checkbox text fields (so, text fields with fewer than 30 options), we track both the full set of options and the filtered set of options. This allows us to display the count and update the count after filtering. We display all of the options (all of the 'originalValues'), even if filtering has eliminated an option from the data view. If an option has been eliminated, we display the count as 0. We get the filtered count from the list of filtered 'values'.